test failures (errors=2) in release 3.1.15; test_list_registered_tasks, test_apply

py2.7 passes all fine.  Under py3s though both these fall short

``` python
 * python3_3: running distutils-r1_run_phase python_test
EE......../usr/lib64/python3.3/unittest/case.py:942: DeprecationWarning: assertDictContainsSubset is deprecated
  DeprecationWarning)

........................................................
======================================================================
ERROR: test_list_registered_tasks (djcelery.tests.test_views.test_registered_tasks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.3/site-packages/anyjson/__init__.py", line 87, in dumps
    return self._encode(data)
  File "/usr/lib64/python3.3/site-packages/simplejson/__init__.py", line 366, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib64/python3.3/site-packages/simplejson/encoder.py", line 269, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.3/site-packages/simplejson/encoder.py", line 348, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib64/python3.3/site-packages/simplejson/encoder.py", line 246, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: dict_keys([]) is not JSON serializable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/djcelery/tests/test_views.py", line 122, in test_list_registered_tasks
    json = self.client.get(registered_tasks())
  File "/usr/lib64/python3.3/site-packages/django/test/client.py", line 473, in get
    response = super(Client, self).get(path, data=data, **extra)
  File "/usr/lib64/python3.3/site-packages/django/test/client.py", line 280, in get
    return self.request(**r)
  File "/usr/lib64/python3.3/site-packages/django/test/client.py", line 444, in request
    six.reraise(*exc_info)
  File "/usr/lib64/python3.3/site-packages/django/utils/six.py", line 550, in reraise
    raise value
  File "/usr/lib64/python3.3/site-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/djcelery/views.py", line 82, in registered_tasks
    'periodic': tasks.periodic().keys()})
  File "/mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/djcelery/views.py", line 20, in JsonResponse
    return HttpResponse(serialize(response), content_type='application/json')
  File "/usr/lib64/python3.3/site-packages/anyjson/__init__.py", line 141, in dumps
    return implementation.dumps(value)
  File "/usr/lib64/python3.3/site-packages/anyjson/__init__.py", line 89, in dumps
    raise TypeError(TypeError(*exc.args)).with_traceback(sys.exc_info()[2])
  File "/usr/lib64/python3.3/site-packages/anyjson/__init__.py", line 87, in dumps
    return self._encode(data)
  File "/usr/lib64/python3.3/site-packages/simplejson/__init__.py", line 366, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib64/python3.3/site-packages/simplejson/encoder.py", line 269, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.3/site-packages/simplejson/encoder.py", line 348, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib64/python3.3/site-packages/simplejson/encoder.py", line 246, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: dict_keys([]) is not JSON serializable

======================================================================
ERROR: test_apply (djcelery.tests.test_views.test_task_apply)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/djcelery/tests/test_views.py", line 95, in test_apply
    task_apply(kwargs={'task_name': mytask.name}) + '?x=4&y=4',
  File "/usr/lib64/python3.3/site-packages/django/test/client.py", line 473, in get
    response = super(Client, self).get(path, data=data, **extra)
  File "/usr/lib64/python3.3/site-packages/django/test/client.py", line 280, in get
    return self.request(**r)
  File "/usr/lib64/python3.3/site-packages/django/test/client.py", line 444, in request
    six.reraise(*exc_info)
  File "/usr/lib64/python3.3/site-packages/django/utils/six.py", line 550, in reraise
    raise value
  File "/usr/lib64/python3.3/site-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/djcelery/views.py", line 55, in apply
    return task_view(task)(request)
  File "/mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/djcelery/views.py", line 37, in _applier
    kwargs = dict(((k, v) for k, v in kwargs.iteritems()), **options)
AttributeError: 'QueryDict' object has no attribute 'iteritems'

----------------------------------------------------------------------
Ran 66 tests in 0.696s

FAILED (errors=2)
nosetests /mnt/gen2/TmpDir/portage/dev-python/django-celery-3.1.15/work/django-celery-3.1.15/tests/../djcelery/tests  --cover3-package=djcelery --cover3-branch --cover3-exclude=djcelery,djcelery.tests.*,djcelery.management.*,djcelery.contrib.* --verbosity=1
Creating test database for alias 'default'...
Destroying test database for alias 'default'...
```

These are 2 distinct issues.  The json serialization appears not to be from the pythonic json package not being installed since there are 2 it tries.  Here are some clues (from another dev here)

 Although, the failure due to iteritems is a translation from 2 to 3 problem.
 In python-3 dict.keys() returns a keys iterator rather than a list like it used to
So that could very well be it.
 Just needs a drop on iter for the second test and a cast of keys on the first.
Should be a two line patch. (1 per issue)
These amount to methods to make a fix which need translating into python code py3 style.

Then again, did your local systems and travis declare them all good?

I don't get the test_list_registered_tasks error, however I do get two test fails still with Python 3.4 and django-celery 3.1.17:
 

```
======================================================================
ERROR: test_apply (djcelery.tests.test_views.test_task_apply)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/brian/tree/debian/python-modules/django-celery/tests/../djcelery/tests/test_views.py", line 95, in test_apply
    task_apply(kwargs={'task_name': mytask.name}) + '?x=4&y=4',
  File "/usr/lib/python3/dist-packages/django/test/client.py", line 470, in get
    **extra)
  File "/usr/lib/python3/dist-packages/django/test/client.py", line 286, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/usr/lib/python3/dist-packages/django/test/client.py", line 358, in generic
    return self.request(**r)
  File "/usr/lib/python3/dist-packages/django/test/client.py", line 440, in request
    six.reraise(*exc_info)
  File "/usr/lib/python3/dist-packages/django/utils/six.py", line 659, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/brian/tree/debian/python-modules/django-celery/tests/../djcelery/views.py", line 55, in apply
    return task_view(task)(request)
  File "/home/brian/tree/debian/python-modules/django-celery/tests/../djcelery/views.py", line 37, in _applier
    kwargs = dict(((k, v) for k, v in kwargs.iteritems()), **options)
AttributeError: 'QueryDict' object has no attribute 'iteritems'

======================================================================
FAIL: test_discovery_with_broken (djcelery.tests.test_discovery.TestDiscovery)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/brian/tree/debian/python-modules/django-celery/tests/../djcelery/tests/test_discovery.py", line 45, in test_discovery_with_broken
    self.assertEqual(log, [])
AssertionError: Lists differ: [<warnings.WarningMessage object at 0x7f2b03474c88>] != []

First list contains 1 additional elements.
First extra element 0:
{message : ResourceWarning("unclosed file <_io.TextIOWrapper name='/home/brian/tree/debian/python-modules/django-celery/tests/someapp/tasks.py' mode='r' encoding='utf-8'>",), category : 'ResourceWarning', filename : '/home/brian/tree/debian/python-modules/django-celery/tests/../djcelery/loaders.py', lineno : 204, line : None}

- [<warnings.WarningMessage object at 0x7f2b03474c88>]
+ []
```

--- a/djcelery/loaders.py
+++ b/djcelery/loaders.py
@@ -196,7 +196,8 @@ def find_related_module(app, related_name):
         return
 
     try:
-        imp.find_module(related_name, app_path)
+        file, _, _ = imp.find_module(related_name, app_path)
+        file.close()
     except ImportError:
         return
 

--- a/djcelery/views.py
+++ b/djcelery/views.py
@@ -34,7 +34,7 @@ def task_view(task):
         kwargs = kwdict(request.method == 'POST' and
                         request.POST or request.GET)
         # no multivalue
-        kwargs = dict(((k, v) for k, v in kwargs.iteritems()), **options)
+        kwargs = dict(((k, v) for k, v in kwargs.items()), **options)
         result = task.apply_async(kwargs=kwargs)
         return JsonResponse({'ok': 'true', 'task_id': result.task_id})
 
@@ -78,8 +78,8 @@ def task_status(request, task_id):
 
 def registered_tasks(request):
     """View returning all defined tasks as a JSON object."""
-    return JsonResponse({'regular': tasks.regular().keys(),
-                         'periodic': tasks.periodic().keys()})
+    return JsonResponse({'regular': list(tasks.regular().keys()),
+                         'periodic': list(tasks.periodic().keys())})
 
 
 def task_webhook(fun):

